src/main/java/org/wildfly/security/sasl/digest/AbstractDigestMechanism.java

```diff
@@ -300,6 +300,9 @@ protected DigestWrapper(boolean confidential) {
         byte[] cipheredPart = null;
         try {
             cipheredPart = wrapCipher.update(toCipher);
+            if (cipheredPart == null){
```

Comment:
I think this is still going to be triggering a throw -> catch -> throw.

The throw should be outside the try / catch block so it is not accidentally caught.